test/net/sourceforge/kolmafia/preferences/PreferencesTest.java

```diff
@@ -559,6 +579,130 @@ public void exerciseIsPerUserGlobalProperty() {
     assertTrue(Preferences.isPerUserGlobalProperty("getBreakfast.PreferencesTestFakeUser"));
   }
 
+  public class timeinThread extends Thread {
+    public timeinThread(String s) {
+      super(s);
+    }
+
+    public void run() {
+      LoginManager.timein(USER_NAME);
+    }
+  }
+
+  public static class incrementThread extends Thread {
+    public incrementThread(String s) {
+      super(s);
+    }
+
+    public void run() {
+      int priorValue = Preferences.getInteger("counter");
+      boolean checkIncrementedValue = false;
+
+      Preferences.increment("counter", 1);
+      int postValue = Preferences.getInteger("counter");
+
+      if (postValue != (priorValue + 1)) {
+        // The end result is correct, so I want to note this, but I'm not sure why the read is
+        // wrong.
+        if (checkIncrementedValue) {
+          System.out.println(
+              "Failure: Prior Value: " + priorValue + " Incremented Value: " + postValue);
+          if (postValue == 0) {
+            System.out.println("Failure: value is now 0");
+          }
+        }
+      }
+    }
+  }
+
+  @Test
+  public void timeinDoesNotCauseRaceCondition() {
+    String unrelatedPref = "coalmine";
+    String unrelatedValue = "canary";
+    String incrementedPref = "counter";
+    int threadCount = 100;
+
+    var cleanups =
+        new Cleanups(
+            withSavePreferencesToFile(),
+            withProperty(unrelatedPref, unrelatedValue),
+            withProperty(incrementedPref, 0));
+    try (cleanups) {
+      Thread[] incrementThreads = new Thread[threadCount];
+      Thread timein = new timeinThread("Timein");
+      timein.start();
+
+      IntStream.range(0, threadCount)
+          .forEach(
+              i -> {
+                incrementThreads[i] = new incrementThread("Increment-" + i);
+                incrementThreads[i].start();
+              });
+      try {
+        if (timein.isAlive()) {
+          timein.join(4000);
+        }
+      } catch (InterruptedException ex) {
+        ex.printStackTrace();
+      }
+      if (timein.isAlive()) {
+        System.out.println("Undead thread: " + timein.getName());
+      }
+      IntStream.range(0, threadCount)
+          .forEach(
+              j -> {
+                try {
+                  incrementThreads[j].join(4000);
+                } catch (InterruptedException e) {
+                  e.printStackTrace();
+                }
+                assertFalse(
+                    incrementThreads[j].isAlive(),
+                    "Undead thread: " + incrementThreads[j].getName());
+              });
+
+      assertEquals(
+          unrelatedValue,
+          Preferences.getString(unrelatedPref, false),
+          "unrelated pref does not match");
+      System.out.println("Final value: " + Preferences.getInteger(incrementedPref));
+    }
+  }
+
+  @Disabled
```

Comment:
Does this test still fail?

Reply:
with a minor tweak, no!  I re-enabled it.  It was comparing "counterTwo" to "counter", which was never going to work...